### ISSUE TITLE
Delta time window: start counting from 'fromdate'

### ISSUE THREAD

**Issue by nikmagini** [dmwm/PhedexReplicaMonitoring#15]
For delta aggregation:

--fromdate fromdate --todate todate --interval N

Currently, PBR will select the following days in the [fromdate, todate] time window to calculate deltas:

fromdate-1+N, fromdate-1+2N, fromdate-1+3N, ..., todate

It would be more intuitive to get the following days instead:

fromdate, fromdate+N, fromdate+2N, fromdate+3N, ..., todate

**Comment by vkuznet** [comment]
Nicolo, let's put this into ranges. Does it mean that you want to see
```
[fromdate, fromdate+N-1], [fromdate+N, fromdate+2N-1], [fromdate+2N, fromdate+3N-1], etc.
```
or
```
[fromdate, fromdate+N], [fromdate+N+1, fromdate+2N], [fromdate+2N+1, fromdate+3N]
```
or
```
[fromdate, fromdate+N], [fromdate+N, fromdate+2N], [fromdate+2N, fromdate+3N]
```
so the real question is how to define boundaries, rather how to start.

**Comment by nikmagini** [comment]
Hi,

Thinking in term of ranges, I think that option 3 makes more sense:

```
[fromdate, fromdate+N], [fromdate+N, fromdate+2N], [fromdate+2N, fromdate+3N]
```

And then you calculate deltas at each boundary: fromdate (initial value), fromdate+N, fromdate+2N, etc.

**Comment by nikmagini** [comment]
(of course this confusion comes from the fact that times are really a continuous value, but we only have a sampling at discrete daily intervals...)

**Comment by vkuznet** [comment]
But this will break with 1m interval where you wanted each start element to be
first day of the month while last element be a last day of the month.

If we want consistency with ranges, then we should use

```
[fromdate, fromdate+N-1], [fromdate+N, fromdate+2N-1], [fromdate+2N, fromdate+3N-1]
```

this will work for any given N where N is numeric and then if user supply
1m, then month will define a month within one range where new range will start
with first day of the month.

**Comment by nikmagini** [comment]
right, I was inconsistent, sorry...

your proposal makes sense.

The important part is that we use the first day in each range to calculate deltas then

**Comment by vkuznet** [comment]
ahh, we're both wrong. The consistent results are
```
[fromdate, fromdate+N], [fromdate+N+1, fromdate+2N], [fromdate+2N+1, fromdate+3N]
```

if N=1m and we start from 2016-01-01 we'll get
```
[2016-01-01, 2016-01-31], [2016-02-01, 2016-02-29]
```
while if we have N=31 we get:
```
[2016-01-01, 2016-01-31], [2016-02-01, 2016-03-02]
```

**Comment by vkuznet** [comment]
I implemented discussed interval range, see PR #16. Closing the ticket.
